reviewrot/gitlabstack.py

```diff
@@ -0,0 +1,170 @@
+                         user(group) name and project name
+        """
+        log.debug('Looking for merge requests for %s -> %s',
+                  uname, project.name)
+
+        # get list of open merge requests for a given repository(project)
+        merge_requests = project.mergerequests.list(project_id=project.id,
+                                                    state='opened')
+        if not merge_requests == 0:
+            log.debug('No open merge requests found for %s/%s ',
+                      uname, project.name)
+        res_ = []
+        for mr in merge_requests:
+            try:
+                mr_date = datetime.datetime.strptime(
+                    mr.created_at, '%Y-%m-%dT%H:%M:%S.%fZ')
+            except ValueError:
+                mr_date = datetime.datetime.strptime(
+                    mr.created_at, '%Y-%m-%dT%H:%M:%SZ')
+            """
+            find the relative time difference between now and
+            pull request filed to retrieve relative information
+            """
+            rel_diff = relativedelta(datetime.datetime.now(),
+                                     mr_date)
+            """
+            find the absolute time difference between now and
+            pull request filed to retrieve absolute information
+            """
+            abs_diff = datetime.datetime.now() - mr_date
+            if (state_ is not None and value is not None and
+                    duration is not None):
+                """
+                check if pull request is older/newer than specified time
+                interval
+                """
+                result = self.check_request_state(abs_diff, rel_diff, state_,
+                                                  value, duration)
+                # skip the request if it doesn't match the specified criteria
+                if not result:
+                    log.debug("merge request '%s' is not %s than specified"
+                              " time interval", mr.title, state_)
+                    continue
+            # format the time interval pull request has been filed since
+            time = self.format_duration(rel_diff)
+            # fetch and format comments for pull request
+            comments = []
+            if(mr.user_notes_count == 1):
+                comments.append('%s' % ', with 1 comment')
+            elif(mr.user_notes_count > 1):
+                comments.append('%s %s %s' % (', with',
+                                              mr.user_notes_count,
+                                              'comments'))
+            comments = ' '.join(comments)
+            # format and print the resultant pull request string
+            res = GitlabReview(user=str(mr.author.username),
+                               title=str(mr.title),
+                               url=str(mr.web_url),
+                               time=time,
+                               comments=comments)
```

Comment:
I think the use of `comments` is a bit complex for what it contains (it also doesn't need to be a list when you first create it). All you are effectively doing is formatting the number of comments into a user friendly string. I believe that should live inside `BaseReview.__str__`. You could just pass `mr.user_notes_count` to the `comments` arg of `GitlabReview`. This change should be reflected across all other classes extending from `BaseService`.